libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/navigation/NavigationMetricsWrapper.java

```diff
@@ -92,52 +93,61 @@ static void rerouteEvent(SessionState sessionState, RouteProgress routeProgress,
-      (int) routeProgress.distanceRemaining(),
-      (int) routeProgress.durationRemaining(),
-      sessionState.secondsSinceLastReroute(), TelemetryUtils.buildUUID(),
-      routeProgress.directionsRoute().getGeometry(), sessionState.mockLocation(),
-      null, null, sessionState.originalGeometry(),
-      sessionState.originalDistance(), sessionState.originalDuration(), null,
-      upcomingInstruction, upcomingType, upcomingModifier, upcomingName, previousInstruction,
-      previousType, previousModifier,
-      previousName, (int) routeProgress.currentLegProgress().currentStep().getDistance(),
-      (int) routeProgress.currentLegProgress().currentStep().getDuration(),
-      (int) routeProgress.currentLegProgress().currentStepProgress().distanceRemaining(),
-      (int) routeProgress.currentLegProgress().currentStepProgress().durationRemaining(),
-      sessionState.currentStepCount(), sessionState.originalStepCount()
+        sdkIdentifier, BuildConfig.MAPBOX_NAVIGATION_VERSION_NAME, sessionState.sessionIdentifier(),
+        location.getLatitude(), location.getLongitude(),
+        sessionState.currentGeometry(), "unknown",
+        (int) routeProgress.directionsRoute().getDistance(),
+        (int) routeProgress.directionsRoute().getDuration(),
+        sessionState.rerouteCount(), sessionState.startTimestamp(), beforeLocations, afterLocations,
+        (int) sessionState.routeProgressBeforeReroute().distanceTraveled(),
+        (int) sessionState.routeProgressBeforeReroute().distanceRemaining(),
+        (int) sessionState.routeProgressBeforeReroute().durationRemaining(),
+        (int) routeProgress.distanceRemaining(),
+        (int) routeProgress.durationRemaining(),
+        sessionState.secondsSinceLastReroute(), TelemetryUtils.buildUUID(),
+        routeProgress.directionsRoute().getGeometry(), sessionState.mockLocation(),
+        null, null, sessionState.originalGeometry(),
+        sessionState.originalDistance(), sessionState.originalDuration(), null,
+        upcomingInstruction, upcomingType, upcomingModifier, upcomingName, previousInstruction,
+        previousType, previousModifier,
+        previousName, (int) routeProgress.currentLegProgress().currentStep().getDistance(),
+        (int) routeProgress.currentLegProgress().currentStep().getDuration(),
+        (int) routeProgress.currentLegProgress().currentStepProgress().distanceRemaining(),
+        (int) routeProgress.currentLegProgress().currentStepProgress().durationRemaining(),
+        sessionState.currentStepCount(), sessionState.originalStepCount()
     ));
   }
 
+  private static Location[] prepareLocations(List<Location> locations) {
+    // Check if the list of locations is empty and if so return an empty array
+    if (locations == null || locations.isEmpty()) {
+      return new Location[0];
+    }
+    // Reverse the list order to conform with the spec
+    Collections.reverse(locations);
```

Comment:
Is this still necessary since locations are added with `addLast()` instead of `push()`?